apps/src/code-studio/components/libraries/LibraryPublisher.jsx

```diff
@@ -0,0 +1,248 @@
+    onUnpublishSuccess: PropTypes.func.isRequired,
+    libraryDetails: PropTypes.object.isRequired,
+    clientApi: PropTypes.object.isRequired
+  };
+
+  state = {
+    publishState: PublishState.DEFAULT,
+    libraryName: libraryParser.suggestName(
+      this.props.libraryDetails.libraryName
+    ),
+    libraryDescription: this.props.libraryDetails.libraryDescription,
+    selectedFunctions: this.props.libraryDetails.selectedFunctions
+  };
+
+  setLibraryName = event => {
+    let sanitizedName = libraryParser.sanitizeName(event.target.value);
+    if (sanitizedName === this.state.libraryName) {
+      return;
+    }
+    this.setState({libraryName: sanitizedName});
+  };
+
+  publish = () => {
+    let {libraryDescription, libraryName, selectedFunctions} = this.state;
+    let {librarySource, sourceFunctionList} = this.props.libraryDetails;
+    let functionsToPublish = sourceFunctionList.filter(sourceFunction => {
+      return selectedFunctions[sourceFunction.functionName];
+    });
+
+    if (!(libraryDescription && functionsToPublish.length > 0)) {
+      this.setState({publishState: PublishState.INVALID_INPUT});
+      return;
+    }
+
+    let libraryJson = libraryParser.createLibraryJson(
+      librarySource,
+      functionsToPublish,
+      libraryName,
+      libraryDescription
+    );
+
+    this.props.clientApi.publish(
+      libraryJson,
+      error => {
+        console.warn(`Error publishing library: ${error}`);
+        this.setState({publishState: PublishState.ERROR_PUBLISH});
+      },
+      () => {
+        this.props.onPublishSuccess(this.state.libraryName);
+      }
+    );
+    dashboard.project.setLibraryName(this.state.libraryName);
+    dashboard.project.setLibraryDescription(libraryDescription);
+  };
+
+  displayNameInput = () => {
+    return (
+      <div>
+        <input
+          style={styles.textInput}
+          type="text"
+          value={this.state.libraryName}
+          onChange={this.setLibraryName}
+          onBlur={event =>
+            this.setState({
+              libraryName: libraryParser.suggestName(event.target.value)
+            })
+          }
+        />
+        <div style={styles.info}>{i18n.libraryNameRequirements()}</div>
+      </div>
+    );
+  };
+
+  resetErrorMessage = () => {
+    if (
+      this.state.libraryDescription &&
+      Object.values(this.state.selectedFunctions).find(value => value) &&
+      this.state.publishState === PublishState.INVALID_INPUT
+    ) {
+      this.setState({publishState: PublishState.DEFAULT});
+    }
+  };
+
+  displayDescription = () => {
+    return (
+      <textarea
+        rows="2"
+        cols="200"
+        style={{...styles.textInput, ...styles.description}}
+        placeholder={i18n.libraryDescriptionPlaceholder()}
+        value={this.state.libraryDescription}
+        onChange={event =>
+          this.setState({libraryDescription: event.target.value})
+        }
+        onBlur={this.resetErrorMessage}
```

Comment:
nit: move `this.resetErrorMessage` to the `onChange` event

Reply:
✔️